@here/harp-mapview/lib/MapView.ts

```diff
@@ -2592,18 +2585,77 @@ export class MapView extends THREE.EventDispatcher {
         this.m_screenCollisions.update(width, height);
 
         this.m_pixelToWorld = undefined;
+        this.m_fog.update(this, this.m_viewRanges.maximum);
+
+        this.updateLookAtSettings();
+    }
 
+    /**
+     * Derive the look at settings (i.e. target, zoom, ...) from the current camera.
+     */
+    private updateLookAtSettings() {
         const cameraPitch = MapViewUtils.extractAttitude(this, this.m_camera).pitch;
-        const cameraPosZ = this.getCameraHeightAboveTerrain(TERRAIN_ZOOM_LEVEL);
-        const zoomLevelDistance = cameraPosZ / Math.cos(Math.min(cameraPitch, Math.PI / 3));
-        this.m_zoomLevel = MapViewUtils.calculateZoomLevelFromDistance(this, zoomLevelDistance);
-        this.m_fog.update(this, this.m_viewRanges.maximum);
 
-        const target = MapViewUtils.getWorldTargetFromCamera(this.m_camera, this.projection);
+        //FIXME: For now we keep the old behaviour when terrain is enabled (i.e. use the camera
+        //       height above terrain to deduce the target distance).
+        //       This leads to zoomlevel changes while panning. We have to find a proper solution
+        //       for terrain (e.g. raycast with the ground surfcae that is elevated by the average
+        //       elevation in the scene)
+        const terrainEnabled = this.elevationProvider !== undefined;
+
+        // Even for a tilt of 90° raycastTargetFromCamera is returning some point almost at
+        // infinity.
+        const target =
+            !terrainEnabled && cameraPitch < PITCH_LIMIT_RAD
+                ? this.raycastTargetFromCamera()
+                : null;
         if (target !== null) {
+            // We have a valid target point. Use it to compute the distance.
             this.m_targetWorldPos.copy(target);
-            this.m_targetGeoPos = this.projection.unprojectPoint(target);
+            this.m_targetGeoPos = this.projection.unprojectPoint(this.m_targetWorldPos);
             this.m_targetDistance = this.camera.position.distanceTo(target);
+        } else {
+            // We either reached the [[PITCH_LIMIT]] or we did not hit the ground surface.
+            // In this case we do the reverse, i.e. compute some fallback distance and
+            // use it to compute the tagret point by using the camera direction.
+            const cameraPosZ = this.getCameraHeightAboveTerrain(TERRAIN_ZOOM_LEVEL);
+
+            //For flat projection we fallback to the target distance at 89 degree pitch.
+            //For spherical projection we fallback to the tangent line distance
+            this.m_targetDistance =
+                this.projection.type === ProjectionType.Planar
```

Comment:
Please see my corresponding patchset #1399. Two things I need to mention here:
- 89 degrees will not always return fine tuned target point, because angle to visual horizon highly depends on the camera altitude and at high attitudes it will end-up very far from the map itself.
- for spherical projections we may need some tolerance instead on tangent point, because the rotation (controls) math may break up quickly on the corner case.

Reply:
Maybe to go forward I could integrate the changes from #1399 int this PR?!